DataMigration/MainWindow.xaml.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Windows;
+using System.Configuration;
+using DataMigration.Logger;
+using DataMigration.Logger.Enums;
+using DataMigration.Models;
+using DataMigration.PostgresDB;
+using DataMigration.PostgresDB.Entities;
+using RabbitMQ.Client.Events;
+using Newtonsoft.Json;
+
+namespace DataMigration
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow
+    {
+        private readonly XamlTextBoxLogger _logger;
+        private readonly DataProvider _dataProvider;
+        private readonly DataProvider _dataProviderDocker;
+
+
+        public MainWindow()
+        {
+            InitializeComponent();
+            _logger = new XamlTextBoxLogger(this);
+            _logger.Log("=========== Logger started working! ==========", LogLevel.Info);
+
+            try
+            {
+                _dataProvider = new DataProvider(ConfigurationManager.ConnectionStrings["psqlConnectionString"].ConnectionString);
+                _dataProviderDocker = new DataProvider(ConfigurationManager.ConnectionStrings["psqlDockerConnectionString"].ConnectionString);
+            }
+            catch (Exception e)
+            {
+                _logger.Log($"Error occured while creating data providers: {e.Message}", LogLevel.Error);
+            }
+            
+
+            _dataProvider.LogEventHappened += _logger.Log;
+            _dataProviderDocker.LogEventHappened += _logger.Log;
+
+        }
+
+        private void StartMigratingDataClick(object sender, RoutedEventArgs e)
+        {
+            try
+            {
+                var mqClient = new RabbitMqClient(ConfigurationManager.AppSettings["rabbitMqHost"], ConfigurationManager.AppSettings["rabbitVhost"]);
+                mqClient.LogEventHappened += _logger.Log;
+
+                var amountToGet = int.Parse(ConfigurationManager.AppSettings["objectsPerIteration"]);
+                var data = _dataProvider.GetHistoricalOcrData(amountToGet);
+
+                while (data.Count != 0)
+                {
+                    _logger.Log($"Processing data... Count: {data.Count}", LogLevel.Debug);
+
+                    _logger.Log("Serializing data...", LogLevel.Debug);
+                    var jsonParsedData = data.Select((item) => (JsonConvert.SerializeObject(item))).ToList();
+                    _logger.Log("Serializing successed!", LogLevel.Debug);
+
+                    SendDataToRabbitMq(jsonParsedData, mqClient);
+                    _dataProvider.DeleteData(data);
+
+                    data = _dataProvider.GetHistoricalOcrData(amountToGet);
+                }
+
+                _logger.Log("Consuming messages from Rabbit", LogLevel.Debug);
+                mqClient.ConsumeMessagesFromQueue("data", ReceiveMessage);
```

Comment:
consuming can be started before publishing messages into queue (even during the start of application but you have to be confident in queue existence or create it otherwise)